File: Python/HackerRank/circularArrayRotation.py

def circularArrayRotation(a, k, queries):
    k %= len(a)
    r = a[-k:] + a[:-k]
    return [r[i] for i in queries]

def main():
    n, k, q = map(int, input().split())
    a = list(map(int, input().split()))
    queries = [int(input()) for _ in range(q)]
    print(*circularArrayRotation(a, k, queries), sep="\n")

if __name__ == '__main__':
    main()

File: golang/hackerrank/arrays/circularArrayRotation.go
package arrays



// CircularArrayRotation performs circular rotation and answers queries.
func CircularArrayRotation(a []int, k int, queries []int) []int {
	n := len(a)
	k %= n

	// r = a[-k:] + a[:-k]
	r := make([]int, n)
	copy(r, a[n-k:])
	copy(r[k:], a[:n-k])

	result := make([]int, len(queries))
	for i, q := range queries {
		result[i] = r[q]
	}
	return result
}
